Std/Data/Bool.lean

```diff
@@ -123,11 +123,13 @@ theorem xor_right_comm : ∀ (x y z : Bool), xor (xor x y) z = xor (xor x z) y :
 
 theorem xor_assoc : ∀ (x y z : Bool), xor (xor x y) z = xor x (xor y z) := by decide
 
-@[simp]
-theorem xor_left_inj : ∀ (x y z : Bool), xor x y = xor x z ↔ y = z := by decide
+@[simp] theorem xor_left_inj : ∀ (x y z : Bool), xor x y = xor x z ↔ y = z := by decide
 
-@[simp]
-theorem xor_right_inj : ∀ (x y z : Bool), xor x z = xor y z ↔ x = y := by decide
+@[simp] theorem xor_right_inj : ∀ (x y z : Bool), xor x z = xor y z ↔ x = y := by decide
+
+@[simp] theorem xor_eq_true : ∀ (x y : Bool), (xor x y = true) = ¬(x = true ↔ y = true) := by decide
```

Comment:
I'd omit `simp` from this.  I think `xor x y = true` is a lot more readable in a non-terminal `simp` while this particular rule is not going to consistently lead to a terminal simp.

Reply:
This is for consistency with core's src/Init/SimpLemmas.lean It would be unusual for the analogous `and`, `or` lemmas to be simp but not for `xor`. That said, I was never sure these simp lemmas should actually be simp.